src/plotrecipes.jl

```diff
@@ -248,3 +248,134 @@ function DD.refdims_title(refdim::Band; issingle=false)
     end
 end
 
+
+# Makie.jl recipes
+
+# For now, these are just basic conversions from 2D rasters to surface-like (surface, heatmap) plot types.
+# Once Makie supports figure level recipes, we should integrate those.
+    
+# First, define default plot types for Rasters
+MakieCore.plottype(raw_raster::AbstractRaster{<: Union{Missing, Real}, 1}) = MakieCore.Lines
+MakieCore.plottype(raw_raster::AbstractRaster{<: Union{Missing, Real}, 2}) = MakieCore.Heatmap
+# 3d rasters are a little more complicated - if dim3 is a singleton, then heatmap, otherwise volume
+function MakieCore.plottype(raw_raster::AbstractRaster{<: Union{Missing, Real}, 3})
+    if size(raw_raster, 3) == 1
+        MakieCore.Heatmap
+    else
+        MakieCore.Volume
+    end
+end
+
+
+missing_or_float32(num::Number) = Float32(num)
+missing_or_float32(::Missing) = missing
+
+# then, define how they are to be converted to plottable data
+function MakieCore.convert_arguments(::MakieCore.PointBased, raw_raster::AbstractRaster{<: Union{Missing, Real}, 1})
+    z = map(Rasters._prepare, dims(raw_raster))
+    return (parent(Float32.(replace_missing(missing_or_float32.(raw_raster), missingval = NaN32))), index(z))
+end
+    
+
+function MakieCore.convert_arguments(::MakieCore.ContinuousSurface, raw_raster::AbstractRaster{<: Union{Missing, Real}, 2})
+    raster = replace_missing(missing_or_float32.(raw_raster), missingval = NaN32)
+    ds = DD._fwdorderdims(raster)
+    A = permutedims(raster, ds)
+    x, y = dims(A)
+    xs, ys, zs = DD._withaxes(x, y, (A))
+    return (xs, ys, zs)
+end
+
+function MakieCore.convert_arguments(::MakieCore.DiscreteSurface, raw_raster::AbstractRaster{<: Union{Missing, Real}, 2})
+    raster = replace_missing(missing_or_float32.(raw_raster), missingval = NaN32)
+    ds = DD._fwdorderdims(raster)
+    A = permutedims(raster, ds)
+    x, y = dims(A)
+    xs, ys, zs = DD._withaxes(x, y, (A))
+    return (Makie.edges(xs), Makie.edges(ys), zs)
```

Comment:
@asinghvi17 isn't this the problem? `Makie.edges` is called here and this method is called in the tests where we only have MakieCore

Reply:
Oops - guess I was testing and somehow pulled that in.  Have reimplemented the function.